# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1516403382 28800
#      Fri Jan 19 15:09:42 2018 -0800
# Node ID f39ae754a7d88191a6f0fb248866577133d2413b
# Parent  298c6a6bd869e2e1936492306f0e1b781e02b3a1
configure: test that makeinfo supports the @sortas command

* acinclude.m4 (OCTAVE_PROG_MAKEINFO): Test that makeinfo supports the @sortas
command, print a warning and disable docs if it doesn't.  This makes explicit
the newest command required to build the manual, which depends on Texinfo 6.1
and newer.
* configure.ac: Update disabling documentation warning messages.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2772,30 +2772,30 @@ AC_DEFINE_UNQUOTED([JAVA_LDPATH], ["$JAV
 ### makeinfo are missing.  Skip these warnings if building docs was
 ### intentionally disabled with a configure option.
 
 ENABLE_DOCS=yes
 AC_ARG_ENABLE([docs],
   [AS_HELP_STRING([--disable-docs], [don't build documentation files])],
   [if test "$enableval" = no; then
      ENABLE_DOCS=no
-     warn_docs="building documentation disabled; make dist will fail"
+     warn_docs="building documentation disabled; make dist will fail."
      OCTAVE_CONFIGURE_WARNING([warn_docs])
    fi], [])
 if test $ENABLE_DOCS = yes; then
   if test $opengl_graphics = no || test -n "$warn_OSMesa"; then
     if test -n "$warn_gnuplot"; then
       ENABLE_DOCS=no
-      warn_docs_graphics="building documentation disabled because no suitable graphics toolkit is available; make dist will fail"
+      warn_docs_graphics="building documentation disabled because no suitable graphics toolkit is available; make dist will fail."
         OCTAVE_CONFIGURE_WARNING([warn_docs_graphics])
     fi
   fi
   if test -n "$warn_makeinfo"; then
     ENABLE_DOCS=no
-    warn_docs_makeinfo="building documentation disabled because makeinfo was not found; make dist will fail"
+    warn_docs_makeinfo="building documentation disabled because makeinfo was not found or does not support @sortas; make dist will fail."
     OCTAVE_CONFIGURE_WARNING([warn_docs_makeinfo])
   fi
 fi
 if test $ENABLE_DOCS = yes; then
   AC_DEFINE(ENABLE_DOCS, 1,
     [Define to 1 to build Octave documentation files.])
 fi
 AM_CONDITIONAL([AMCOND_BUILD_DOCS], [test $ENABLE_DOCS = yes])
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2628,16 +2628,54 @@ AC_DEFUN([OCTAVE_PROG_MAKEINFO], [
   if test -z "$MKINFO"; then
     warn_makeinfo="
 
 I didn't find makeinfo, which is required for reading documentation.
 You may install a copy later for Octave to use.
 "
     OCTAVE_CONFIGURE_WARNING([warn_makeinfo])
   fi
+  dnl If we have a GNU makeinfo program, see if it supports the @sortas command
+  dnl for defining a custom sort key for an index entry.
+  if test -n "$MKINFO"; then
+    AC_CACHE_CHECK([for makeinfo support for @sortas command],
+      [octave_cv_makeinfo_sortas_command],
+      [cat << EOF > conftest.texi
+\input texinfo
+@node Top
+@top Document
+@menu
+* Chapter::
+* Index::
+@end menu
+@node Chapter
+@chapter Chapter
+@cindex @sortas{a} foo
+@node Index
+@unnumbered Index
+@printindex cp
+@bye
+EOF
+        if $MKINFO --no-warn conftest.texi 2>/dev/null; then
+          octave_cv_makeinfo_sortas_command=yes
+        else
+          octave_cv_makeinfo_sortas_command=no
+        fi
+        rm -f conftest.info conftest.texi
+    ])
+    if test $octave_cv_makeinfo_sortas_command = no; then
+      warn_makeinfo="
+
+I wasn't able to find a version of GNU makeinfo that supports the
+@sortas command, but it's only a problem if you need to build the
+manual, which is the case if you're building from VCS sources.
+"
+      OCTAVE_CONFIGURE_WARNING([warn_makeinfo])
+    fi
+  fi
 ])
 dnl
 dnl What pager should we use?
 dnl
 AC_DEFUN([OCTAVE_PROG_PAGER], [
   if test "$cross_compiling" = yes; then
     DEFAULT_PAGER=less
     AC_MSG_RESULT([assuming $DEFAULT_PAGER exists on $canonical_host_type host])
